agent/pagebroker/daemon_test.cpp

```
@@ -0,0 +1,194 @@
+  for (const auto& id :
+       {std::string("../escape"), std::string("nested/name"), std::string("nested\\name"),
+        std::string("nul\0suffix", 10)}) {
+    auto abort = RequestFor(id);
+    abort.mutable_abort();
+    const auto response = broker().HandleRequest(abort);
+    EXPECT_TRUE(response.has_failure());
+    EXPECT_EQ(response.failure().code(), Failure::INVALID_REQUEST);
+  }
+}
+
+TEST_F(BrokerTest, RejectsSymlinkInRestoreSource)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)